src/twisted/trial/test/test_tests.py

```diff
@@ -1417,6 +1417,43 @@ def test_generator(self):
             result.errors[0][1].value.args[0],
         )
 
+    def test_errorOnAsyncGeneratorFunction(self):
+        """
+        In a TestCase, a test method which is an async generator function is reported
+        as an error, as such a method will never run assertions.
+        """
+
+        class AsyncGeneratorTestCase(unittest.TestCase):
+            """
+            A fake TestCase for testing purposes.
+            """
+
+            async def test_async_generator(self):
+                """
+                A method which is also a generator function, for testing
+                purposes.
+                """
+                self.fail("this should never be reached")  # pragma: no cover
+                yield
```

Comment:
pragma: no cover needed here too, or maybe on the suite-starting colon, I think coverage reflects that